File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) 
    {
        int max_size, min_size;
        set<int> s,s1;
        vector<int> vec;
        if(nums1.size() > nums2.size())
        {
            for(int i = 0; i < nums1.size(); i++)
            {
                if(s.find(nums1[i]) != s.end())
                    continue;
                s.insert(nums1[i]);
            }
            for(int i = 0; i < nums2.size(); i++)
            {
                if(s.find(nums2[i]) == s.end())
                {
                    continue;
                }
                s1.insert(nums2[i]);
            }
            for (auto it = s1.begin(); it != s1.end(); it++) {
                    vec.emplace_back(*it);
            }
        }   
        else
        {
            for(int i = 0; i < nums2.size(); i++)
            {
                if(s.find(nums2[i]) != s.end())
                    continue;
                s.insert(nums2[i]);

            }
            for(int i = 0; i < nums1.size(); i++)
            {
                if(s.find(nums1[i]) == s.end())
                {
                    continue;
                }
                s1.insert(nums1[i]);
            }
            for (auto it = s1.begin(); it != s1.end(); it++) {
                    vec.emplace_back(*it);
            }
        }
        
        return vec;
    }
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, a: List[int], b: List[int]) -> List[int]:
        a.sort()
        b.sort()
        m = len(a)
        n = len(b)
        res = []
        i = j = 0
        while i < m and j < n:
            if i > 0 and a[i] == a[i-1]:
                i += 1
                continue
            elif a[i] < b[j]:
                i += 1
            elif a[i] > b[j]:
                j += 1
            else:
                res.append(a[i])
                i += 1
                j += 1
        return res